pkg/clustermesh/kvstoremesh/kvstoremesh.go

```diff
@@ -88,6 +94,30 @@ func newKVStoreMesh(lc cell.Lifecycle, params params) *KVStoreMesh {
 	return &km
 }
 
+type SyncWaiterParams struct {
+	cell.In
+
+	KVStoreMesh *KVStoreMesh
+	SyncState   syncstate.SyncState
+	Lifecycle   cell.Lifecycle
+	JobRegistry job.Registry
+	Scope       cell.Scope
+}
+
+func RegisterSyncWaiter(p SyncWaiterParams) {
+	jobGroup := p.JobRegistry.NewGroup(p.Scope)
+	jobGroup.Add(
+		job.OneShot("kvstoremesh-sync-waiter", func(ctx context.Context, health cell.HealthReporter) error {
+			syncedCallback := p.SyncState.WaitForResource()
+			p.SyncState.Stop()
```

Comment:
I think this should be fine, since the syncstate cannot be "complete" until `Stop()` is called on the underlying swg, but I moved it into the main function anyways